### DISCUSSION TITLE
cmd functions not working as expected on windows(cmd.exe)

### DISCUSSION THREAD

**Discussion by LinArcX** [gokcehan/lf#1586] [Q&A] [answered]
on windows 10 and with cmd. i want to use lf. this is my lfrc:

```
set shell cmd

set hidden
#set filesep " "
set info size
set dircounts
set icons
set drawbox
#set ifs "\n"

# <pgup>, <pgdn>, <c-b>, <c-f>, <c-u>, <c-d>, <up>, <down>
map r
map f
map m
map i
map p
map x $$f
map X !$f

## general mappings
map ; read
map . set hidden!
map o open
map <enter>  &notepad.exe

cmd open ${{
  &notepad.exe
}}
```
The problem is that when I press `o`, the `open()` won't call. but as you can see, <enter> do the same thing as `o`. my question is why cmd functions on windows(cmd.exe) doesn't work? is it a bug?

**Comment by joelim-work** [comment]
If you're using Windows CMD, you should take a look at the [example config file](https://github.com/gokcehan/lf/blob/master/etc/lfrc.cmd.example).

> Shell commands with multiline definitions and/or positional arguments and/or quotes do not work in Windows

In other words you can't use multiline `{{ ... }}` constructs.

---

The following `open` command might be what you're looking for:

```
cmd open &notepad.exe %f%
```

**Reply by LinArcX** [reply]
i want to use multiline constructs.

**Reply by joelim-work** [reply, marked as answer]
This isn't possible due to limitations in Windows CMD, see #431 

You will have to either place all the desired commands in a wrapper batch script, or use a different shell like PowerShell.
